layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -588,12 +588,14 @@ dotspacemacs-persistent-server to be t"
 (defun spacemacs/kill-emacs ()
   "Lose all changes and exit Spacemacs"
   (interactive)
+  (run-hook-with-args-until-failure 'kill-emacs-query-functions)
```

Comment:
You're right, I just copied the implementation from `save-buffers-kill-emacs` but I reliaze now that it's wrapped in a big `and`. That was confusing.